File: myPow.js
var myPow = function(x, n) {
  if (!n) {
    return 1;
  }
  if (!x) {
    return 0;
  }
  let res = 1;
  let tmp = n < 0 ? 1 / x : x;
  let times = Math.abs(n);
  for (; times > 0; times /= 2) {
    if (times % 2) {
      res *= tmp;
    }
    tmp = tmp * tmp;
  }
  return res;
};

myPow(2, 10);

File: my_pow.rs
impl Solution {
  pub fn my_pow(x: f64, n: i32) -> f64 {
    if n == 0 {
      return 1.0;
    }
    let flag = n < 0;
    let mut x: f64 = x;
    let mut n: i32 = n;
    let mut res: f64 = 1.0;
    while(n != 0) {
      if n % 2 == 1 {
        res = res * x;
      }
      x = x * x;
      n = n / 2;
    }
    if flag {
      return 1.0 / res;
    }
    res
  }
}